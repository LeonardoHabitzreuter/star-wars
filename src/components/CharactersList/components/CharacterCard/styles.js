import Card from '/ui/Card'
import styled, { css } from 'styled-components'
import { Button, Typography } from '/ui'
import { phone, tablet } from '/styles/theme'

export const StyledCharacter = styled(Card)`
  ${({ hide }) => hide && 'display: none;'}
  box-sizing: border-box;
  height: 150px;
  width: 100%;
  --cols: 3;
  max-width: calc(100% / var(--cols) - 20px);
  margin: 0 10px;
  margin-bottom: 16px;
  box-shadow: 2px 2px 2px 2px rgba(0, 0, 0, 0.14);

  ${tablet(css`
    --cols: 2;
  `)}

  ${phone(css`
    --cols: 1;
    max-width: 100%;
    margin: 10px 0;
  `)}
`

export const Row = styled.div`
  width: 100%;
  margin: 5px 0;
  display: flex;
  justify-content: space-between;
`

export const Characteristic = styled(Typography)`
  width: 50%;
`

export const DetailsButton = styled(Button)`
  align-self:flex-end;

  ${phone(css`
    align-self: center;
  `)}
`
