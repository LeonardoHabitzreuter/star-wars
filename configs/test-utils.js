import * as theme from '../src/styles/theme'
import React from 'react'
import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(
  ui,
  { wrapper: AllTheProviders, ...options }
)

export * from '@testing-library/react'

export { customRender as render }
