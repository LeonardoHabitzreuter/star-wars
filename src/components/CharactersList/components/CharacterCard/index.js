import * as R from 'ramda'
import React from 'react'
import PropTypes from 'prop-types'
import CharacterPage from '/pages/Character'
import { Typography } from '/ui'
import { StyledCharacter, Row, Characteristic, DetailsButton } from './styles'

const mountLink = R.replace(':id', R.__, CharacterPage.route)

const Character = props => (
  <StyledCharacter>
    <Typography bold variant='h5'>{props.name}</Typography>
    <Row>
      <Characteristic bold>Height: <Typography>{props.height}</Typography></Characteristic>
      <Characteristic bold>Mass: <Typography>{props.mass}</Typography></Characteristic>
    </Row>
    <Row>
      <Characteristic bold>Birth year: <Typography>{props.birth_year}</Typography></Characteristic>
      <Characteristic bold>Gender: <Typography>{props.gender}</Typography></Characteristic>
    </Row>
    <DetailsButton
      href={mountLink(props.id)}
      modifier='secondary'
    >
      Details
    </DetailsButton>
  </StyledCharacter>
)

Character.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  height: PropTypes.string,
  mass: PropTypes.string,
  birth_year: PropTypes.string,
  gender: PropTypes.string
}

export default Character
