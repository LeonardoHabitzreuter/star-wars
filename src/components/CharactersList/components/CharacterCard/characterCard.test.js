import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import React from 'react'
import CharacterCard from './index'
import { render } from 'test-utils'

describe('CharacterCard component', () => {
  test('should render properly', () => {
    const { container } = render(
      <CharacterCard
        id='1'
        name='Luke Skywalker'
        height='172'
        mass='77'
        birth_year='19BBY'
        gender='male'
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
