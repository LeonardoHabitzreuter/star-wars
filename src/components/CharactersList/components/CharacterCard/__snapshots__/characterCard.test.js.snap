// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard component should render properly 1`] = `
.c1 {
  font-size: 22px;
  color: #141414;
  margin: 0;
  font-weight: bold;
  line-height: 1.5;
  text-align: left;
}

.c3 {
  font-size: 14px;
  color: #141414;
  margin: 0;
  font-weight: bold;
  line-height: 1.5;
  text-align: left;
}

.c5 {
  font-size: 14px;
  color: #141414;
  margin: 0;
  font-weight: normal;
  line-height: 1.5;
  text-align: left;
}

.c8 {
  border-radius: 2px;
  outline: none;
  cursor: pointer;
  padding: 8px 20px;
  color: white;
  border: solid 1px #000;
  background-color: #545454;
  font-size: 18px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.33;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: center;
}

.c8:active {
  background-color: #2e2e2e;
}

.c6 {
  padding: 8px 20px;
  padding: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 8px;
  border-radius: 4px;
  border: solid 1px #fff;
  box-sizing: border-box;
  height: 150px;
  width: 100%;
  --cols: 3;
  max-width: calc(100% / var(--cols) - 20px);
  margin: 0 10px;
  margin-bottom: 16px;
  box-shadow: 2px 2px 2px 2px rgba(0,0,0,0.14);
}

.c2 {
  width: 100%;
  margin: 5px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  width: 50%;
}

.c7 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

@media (max-width:414px) {
  .c8 {
    padding: 8px 20px;
    width: 100%;
  }
}

@media (max-width:414px) {
  .c6 {
    padding: 8px 20px;
    width: 100%;
  }
}

@media (max-width:1024px) {
  .c0 {
    --cols: 2;
  }
}

@media (max-width:414px) {
  .c0 {
    --cols: 1;
    max-width: 100%;
    margin: 10px 0;
  }
}

@media (max-width:414px) {
  .c7 {
    -webkit-align-self: center;
    -ms-flex-item-align: center;
    align-self: center;
  }
}

<div
  class="sc-fzokOt c0"
>
  <h5
    class="c1"
    color="#141414"
  >
    Luke Skywalker
  </h5>
  <div
    class="c2"
  >
    <span
      class="c3 c4"
      color="#141414"
    >
      Height: 
      <span
        class="c5"
        color="#141414"
      >
        172
      </span>
    </span>
    <span
      class="c3 c4"
      color="#141414"
    >
      Mass: 
      <span
        class="c5"
        color="#141414"
      >
        77
      </span>
    </span>
  </div>
  <div
    class="c2"
  >
    <span
      class="c3 c4"
      color="#141414"
    >
      Birth year: 
      <span
        class="c5"
        color="#141414"
      >
        19BBY
      </span>
    </span>
    <span
      class="c3 c4"
      color="#141414"
    >
      Gender: 
      <span
        class="c5"
        color="#141414"
      >
        male
      </span>
    </span>
  </div>
  <a
    class="c6 c7"
    href="/character/1"
    modifier="secondary"
    type="button"
  >
    <button
      class="c8 c7"
      type="button"
    >
      Details
    </button>
  </a>
</div>
`;
